docker/src/main/java/brooklyn/entity/container/docker/DockerContainerImpl.java

```diff
@@ -614,13 +614,15 @@ public DockerContainerLocation createLocation(Map flags) {
             }
 
             // Create our wrapper location around the container
-            LocationSpec<DockerContainerLocation> spec = LocationSpec.create(DockerContainerLocation.class)
+            LocationSpec<DockerContainerLocation> spec = (LocationSpec<DockerContainerLocation>) LocationSpec.create(DockerContainerLocation.class)
```

Comment:
Changing as suggested, didn't come up with a better way of dealing with the type erasure.